### DISCUSSION TITLE
Generate type safe client for remote schema from apollo server

### DISCUSSION THREAD

**Discussion by mr-nothing** [Netflix/dgs-framework#1749] [Q&A] [answered]
There is a wonderful feature in dgs codegen plugin that allows us to generate dto objects to use it in queries and mutations in a typesafe manner.
But I failed to understand how to generate these objects in case I use apollo server and stitch my application schema from multiple services.
Is there some type of manual or example of how it can be done? Current documentation covers only one-service/local schema cases.
There are such a feature in [apollo kotlin client](https://www.apollographql.com/docs/kotlin/essentials/queries) and I wonder if it is a way to go for netflix dgs users. Or in case of netflix dgs I need to follow completely different approach?

Thank you!

**Comment by srinivasankavitha** [comment, marked as answer]
You will need a way to get the application schema from the introspection endpoint and have it local in order to generate code using the plugin. We have such use cases and we obtain the schema from the introspection endpoint.

**Comment by mr-nothing** [comment]
Ok, thank you! It seems I can just generate it with use of IDEA plugin and put it locally manually or do the same with use of some gradle task
